File: greedy/leetcode_409/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        map = {}
        result = 0
        has_odd = False

        for char in s:
            map[char] = map.get(char, 0) + 1

        for value in map.values():
            if value % 2 == 0:
                result += value
            else:
                result += value - 1
                has_odd = True

        if has_odd:
            result += 1
        
        return result
        

File: greedy/leetcode_409/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function(s) {
    let map = {};
    let result = 0;
    let hasOdd = false;

    for(const char of s) {
        map[char] = (map[char] ?? 0) + 1;
    }

    for(const value of Object.values(map)) {
        if (value % 2 === 0) {
            result += value;
        } else {
            result += value - 1;
            hasOdd = true;
        }
    }

    if(hasOdd) {
        result ++;
    }

    return result;
};